src/components/widget/ActionButton.js

```diff
@@ -215,14 +215,44 @@ class ActionButton extends Component {
     });
   };
 
+  processStatus = (status, option) => {
+    const {
+      onChange,
+      docId,
+      windowType,
+      activeTab,
+      fetchTopActions,
+    } = this.props;
+    const changePromise = onChange(status);
+
+    this.statusDropdown.blur();
+    if (changePromise instanceof Promise) {
```

Comment:
do u know which is the reason we check if `changePromise` is instance of Promise?
I mean, this introduces FUD.

Reply:
I copy/pasted this piece of code from previous functionality.

Reply:
We agreed to allow it like that in this case,
but we set in proptypes that onChange is a function and is required.

Reply:
I gave some thoughts and also some testing and that won't work. You see in https://github.com/metasfresh/metasfresh-webui-frontend/blob/90e83805472fe77ee1b00f2492f6c1d403f5c61d/src/components/widget/RawWidget.js#L274 . -> RawWidget L: 274 that returns also undefined   (Promise.resolve(null)) which is passed to onChange . So forcing the onChange to be required function will not work in this case. I'll find a sol on Monday

Reply:
Fixed with a   onChange: PropTypes.instanceOf(Function)